Apps/Common/src/Services/EmailQueueService.cs

```diff
@@ -128,19 +138,22 @@ public void CloneAndQueue(Guid emailId, bool shouldCommit = true)
         }
 
         /// <inheritdoc />
-        public void QueueNewInviteEmail(string hdid, string toEmail, Uri activationHost)
+        public void QueueNewInviteEmail(string hdid, string toEmail, Uri activationHost, Guid? existingInviteKey)
         {
             Dictionary<string, string> keyValues = new Dictionary<string, string>();
             MessagingVerification invite = new MessagingVerification();
-            invite.InviteKey = Guid.NewGuid();
+            invite.InviteKey = existingInviteKey.HasValue ? existingInviteKey.Value : Guid.NewGuid();
```

Comment:
instead of having the logic here, I think we should move it to the callee to pass a valid non-null guid

Reply:
updated